src/bin/rdcore/rootmap.rs

```diff
@@ -216,9 +207,26 @@ fn get_luks_name(device: &Path) -> Result<String> {
 }
 
 fn get_luks_uuid(device: &Path) -> Result<String> {
-    Ok(runcmd_output!("cryptsetup", "luksUUID", device)?
-        .trim()
-        .into())
+    // The LUKS UUID is a property of the backing block device of *this* block device, so we have
+    // to get its parent. In case of crypt-integrity device we have to recursively go to the
+    // underlying block device
+    let deps = get_blkdev_deps(device)?;
+    match deps.len() {
+        0 => bail!("missing parent device for {}", device.display()),
+        1 => {
+            let device = deps[0].as_path();
```

Comment:
I know this isn't a new issue, but we should be able to match like this instead which is a bit cleaner IMO:

```
match deps.as_slice() {
  [] => bail!("missing parent device for {}", device.display()),
  [device] => { ... }
  _ => bail!("found multiple...")
}
```